eng/common/pipelines/templates/steps/save-package-properties.yml

```diff
@@ -14,9 +14,12 @@ parameters:
   - name: ScriptDirectory
     type: string
     default: eng/common/scripts
+  - name: ServiceDirectoryFilter
+    type: string
+    default: ""
 
 steps:
-  - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+  - ${{ if and(eq(variables['Build.Reason'], 'PullRequest'), or(eq(parameters.ServiceDirectoryFilter, ''), eq(parameters.ServiceDirectory, parameters.ServiceDirectoryFilter))) }}:
```

Comment:
Perhaps I'm not understanding something but I was expecting to see some sort of check for ServiceDirectory == auto in this condition.

Reply:
My is to add the parameter so we can _pass in_ `auto` as the word that we want to filter on. 

![image](https://github.com/user-attachments/assets/a583fedc-8b38-4f63-82a6-7f051d49fce9)

I just implemented the condition wrong 👎

Reply:
Oh. I was misreading the build failure. This condition is working the way I expect it. Given above config, this is what a trigger of `js - core - ci` looks like alongside `js - pullrequest`

![image](https://github.com/user-attachments/assets/104ddee5-0938-455e-808a-925b84874212)

Reply:
Would the ServiceDirectoryFilter ever be anything aside from empty or auto?

Reply:
> Would the ServiceDirectoryFilter ever be anything aside from empty or auto? If not perhaps just make it a boolean.

Realistically no. I'm just avoiding needing to change this code again when someone has an issue for `auto`.

Reply:
Hardcoded to `auto` and removed param.